polaris-core/src/main/java/io/polaris/core/storage/PolarisStorageConfigurationInfo.java

```diff
@@ -171,11 +177,38 @@ public static Optional<PolarisStorageConfigurationInfo> forEntityPath(
                 LOGGER.debug(
                     "Allowing unstructured table location for entity: {}",
                     entityPathReversed.get(0).getName());
-                return configInfo;
+                return userSpecifiedWriteLocations(entityPathReversed.get(0).getPropertiesAsMap())
```

Comment:
It looks like to avoid always doing a storage configuration override we are using the Optional path below, It just feels like it may be simpler to just always add Locs to the Config?

Or we could just do

if (userSpecifiedWriteLocations.notEmpty)
  return Storage Override
else
  configInfo
  
  I'm not sure the functional style here makes it much more readable, One of those things that I think would be much cleaner in Scala.

Reply:
Hmm, yeah I think we can just always add the `StorageConfigurationOverride` with the empty `locs` if not set